backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -766,4 +775,28 @@ public Boolean getTraceEnabled() {
   public Integer getRequestTimeout() {
     return requestTimeout;
   }
+
+  public void setRoleService(RoleService roleService) {
+    this.roleService = roleService;
+  }
+
+  public RoleService getRoleService() {
+    return roleService;
+  }
+
+  public boolean checkConnectionsBasedOnPrivileges(
```

Comment:
can we move this in RoleService?
We should conditionally call this rather than have conditional execution within the function